packages/go/ein/azure.go

```diff
@@ -1854,6 +1854,100 @@ func ConvertAzureRoleEligibilityScheduleInstanceToRel(instance models.RoleEligib
 	return relationships
 }
 
+// ConvertAzureRoleManagementPolicyAssignment will create, or update the properties if it exists, an AZRole of a tenant
+// with the supplied RoleManagementPolicyAssignment properties
+// If EndUserAssignmentGroupApprovers contains GUIDs: an edge will be created from each group to the created AZRole
+// If EndUserAssignmentUsersApprovers contains GUIDs: an edge will be created from each user to the created AZRole
+// If both lists are empty: an edge will be created from the tenant's PrivilegedRoleAdministratorRole to the created AZRole
+func ConvertAzureRoleManagementPolicyAssignment(policyAssignment models.RoleManagementPolicyAssignment) (IngestibleNode, []IngestibleRelationship) {
+	var (
+		rels             = make([]IngestibleRelationship, 0)
+		combinedObjectId = strings.ToUpper(strings.Join([]string{policyAssignment.RoleDefinitionId, policyAssignment.TenantId}, "@"))
+	)
+
+	// Format the incoming user and group ids to uppercase string before creating our nodes
+	for i, _ := range policyAssignment.EndUserAssignmentGroupApprovers {
+		policyAssignment.EndUserAssignmentGroupApprovers[i] = strings.ToUpper(policyAssignment.EndUserAssignmentGroupApprovers[i])
+	}
+
+	for i, _ := range policyAssignment.EndUserAssignmentUserApprovers {
+		policyAssignment.EndUserAssignmentUserApprovers[i] = strings.ToUpper(policyAssignment.EndUserAssignmentUserApprovers[i])
+	}
+
+	// We will want to create or update any existing AZRole node that matches the combinedObjectId
+	// If the node exists, we want to add the new properties to the node
+	targetAZRole := IngestibleNode{
+		ObjectID: combinedObjectId,
+		Label:    azure.Role,
+		PropertyMap: map[string]any{
+			azure.RoleDefinitionId.String():                                  strings.ToUpper(policyAssignment.RoleDefinitionId),
+			azure.TenantID.String():                                          strings.ToUpper(policyAssignment.TenantId),
+			azure.EndUserAssignmentRequiresApproval.String():                 policyAssignment.EndUserAssignmentRequiresApproval,
+			azure.EndUserAssignmentRequiresCAPAuthenticationContext.String(): policyAssignment.EndUserAssignmentRequiresCAPAuthenticationContext,
+			azure.EndUserAssignmentUserApprovers.String():                    policyAssignment.EndUserAssignmentUserApprovers,
+			azure.EndUserAssignmentGroupApprovers.String():                   policyAssignment.EndUserAssignmentGroupApprovers,
+			azure.EndUserAssignmentRequiresMFA.String():                      policyAssignment.EndUserAssignmentRequiresMFA,
+			azure.EndUserAssignmentRequiresJustification.String():            policyAssignment.EndUserAssignmentRequiresJustification,
+			azure.EndUserAssignmentRequiresTicketInformation.String():        policyAssignment.EndUserAssignmentRequiresTicketInformation,
+		},
+	}
+
+	if !policyAssignment.EndUserAssignmentRequiresApproval {
+		// We cannot create the edge or nodes if the assignment does not require approval
```

Comment:
This comment is misleading as we do create the node regardless

Reply:
Whoops, leftover from when I misread the ticket and left everything in an if/elif block